kolibri/plugins/learn/assets/src/views/TopicsPage/TopicRow.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+
+  <div>
+    <!-- header link to folder -->
+    <h2>
+      <KRouterLink
+        :text="topic.title"
+        :to="genContentLink(topic)"
+        class="folder-header-link"
+        :appearanceOverrides="{ color: $themeTokens.text }"
+      >
+        <template #iconAfter>
+          <KIcon
+            icon="chevronRight"
+            :style="{ top: '4px' }"
+          />
+        </template>
+      </KRouterLink>
+    </h2>
+    <!-- card grid of items in folder -->
+    <LibraryAndChannelBrowserMainContent
+      v-if="topic.children && topic.children.length"
+      data-test="children-cards-grid"
+      :contents="topic.children"
+      currentCardViewStyle="card"
+      :gridType="1"
+      @toggleInfoPanel="$emit('toggleInfoPanel', $event)"
+    />
+    <KButton
+      v-if="topic.showMore"
+      class="more-after-grid"
+      data-test="more-button"
+      appearance="basic-link"
+      @click="$emit('showMore', topic.id)"
+    >
+      {{ $tr('showMore') }}
+    </KButton>
+    <KRouterLink v-else-if="topic.viewAll" class="more-after-grid" :to="topic.viewAll">
+      {{ $tr('viewAll') }}
+    </KRouterLink>
+    <KButton
+      v-else-if="topic.viewMore && topic.id !== subTopicLoading"
+      class="more-after-grid"
+      appearance="basic-link"
+      @click="$emit('loadMoreInSubtopic', topic.id)"
+    >
+      {{ coreString('viewMoreAction') }}
+    </KButton>
+
+    <KCircularLoader v-if="topic.id === subTopicLoading" />
+  </div>
+
+</template>
+
+
+<script>
+
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import genContentLink from '../../utils/genContentLink';
+  import LibraryAndChannelBrowserMainContent from '../LibraryAndChannelBrowserMainContent';
+
+  export default {
+    name: 'TopicRow',
```

Comment:
Is this really just a row, or a topic subsection?

Reply:
It's a complicated row :smile: - subsection makes sense to me